routers/web/admin/hooks.go

```diff
@@ -61,13 +62,21 @@ func DefaultOrSystemWebhooks(ctx *context.Context) {
 
 // DeleteDefaultOrSystemWebhook handler to delete an admin-defined system or default webhook
 func DeleteDefaultOrSystemWebhook(ctx *context.Context) {
+	defer ctx.JSON(http.StatusOK, map[string]interface{}{
+		"redirect": setting.AppSubURL + "/admin/hooks",
+	})
+
+	hook, err := webhook.GetSystemOrDefaultWebhook(ctx, ctx.FormInt64("id"))
+	if err != nil {
+		ctx.Flash.Error("GetWebhookByOwnerID: " + err.Error())
+		return
+	}
+
 	if err := webhook.DeleteDefaultSystemWebhook(ctx, ctx.FormInt64("id")); err != nil {
 		ctx.Flash.Error("DeleteDefaultWebhook: " + err.Error())
 	} else {
+		audit.Record(audit.SystemWebhookRemove, ctx.Doer, nil, hook, "Removed webhook %s.", hook.URL)
```

Comment:
Is `"Removed webhook %s."` friendly to audit log readers? It seems a human-oriented message, difficult to be parsed by programs, eg: how to get the real URL by a program parser?

Maybe it could follow the structed logger ?

Reply:
Yes, good idea. Structure logger is more similar as database which is also easier to parse by other tools.

Reply:
The message is less relevant. The `hook` object is passed into the logging. The `WebHook` type gets resolved as
```
TypeDescriptor{Type: "webhook", PrimaryKey: <hook.ID>, FriendlyName: <hook.URL>}
```
It is already structured for every log appender.

Reply:
I still think this design is not general or flexible.

For example, the "push mirror", how do you record the URL/Address ?

Reply:
See https://github.com/go-gitea/gitea/pull/24257#issuecomment-1701195354 for how an entry appears in the file log. From there you can query all info you need I think. I don't think it's desired to dump more or less full objects.

Reply:
But isn't `URL` is a very important "audited" value when adding a push mirror? If it doesn't appear in the audit log, how to audit .....

Reply:
The push mirror is a little bit special because we do not store the url in the database and so there is no easy way to retrieve it in the logger. I can create a PR which adds the url to the database, then I can use it as `FriendlyName`.

Reply:
Then it comes to the question above: is such mechanism general/flexible enough? The "push mirror address" is only one case, there might be more similar cases in current code and in the future. How to make the audit framework have the ability to handle these cases, maybe sometimes it needs to record more than one value in one audit log.

IMO the answer could still be something like the structured logger?

Reply:
The full object is always passed in the event. If more info is needed it can be gathered there. If a new object should be handled, you need to answer the question "What describes this object?" and that's the `FriendlyName`. Usually that's enough to get an overview when reading the logs. If more info about an object is needed, you need to query that info by the `PrimaryKey`.